articles/kotlin-ranges/index.md

```diff
@@ -0,0 +1,80 @@
+
+<iframe height="400px" width="100%" src="https://repl.it/@Linusmuema/Ranges?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>
+
+### 1. To create a progression
+We can use the range operator to create an iterable range. A loop is created that goes through each item from the start value to the end value. This helps us to be sure of the number of iterations in the loop. It gives the developer more control over the loop. In the repl, call the `printNumRange` function and run. The output should be
+
+```bash
+1,2,3,4,5,
+```
+
+This loop goes from 1 to 5. But that is not the only trick ranges have. We can also create ranges using characters. We can create a range that starts from a specific letter in the alphabet to another. This can be from `a` to `e`. You can go ahead and call the `printCharRange` function. The expected output is
+
+```bash
+a,b,c,d,e,
+```
+It was not easy to do this in java. When implemented it led to lots of boilerplate and dirty code.
+
+### 2. To create reverse loops
+It was always a challenge to create a reverse loop as you would interchange values in the old way. This could lead to confusion and take time to debug. An example of a reverse loop in java would be:
+
+```Java
+for (int i = 5; i >= 0; i--) {}
+```
+
+Ranges have a function called `downTo` which can be used to create a reverse progression. We can use it to loop from the end value backwards to the start point. Once you run the `reverseNumRange`, the output should be
+
+```bash
+5,4,3,2,1,
+```
+
+The same applies to the character range. It loops downwards to the start character. The function `reverseCharRange` does that.
+
+```bash
+e,d,c,b,a,
+```
+
+### 3. To generate random numbers
+Sometimes you may want to generate random numbers within certain limits. This is virtually impossible in the old loop. A good option is to use `Math` class and call the `random` function. This however is limiting to the values it returns. It returns double values ranging from 0.0 to 1.0. In addition to that, you needed to add more calculations. It may be confusing to a developer who is not very experienced in Java. For example, to generate random numbers between 1 to 10:
+
+```Java
+int max = 10;
+int min = 1;
+int range = max - min + 1;
+int random = (Math.random() * range) + min;
+```
+
+With ranges, it is as simple as creating the desired range and calling the `random` method. This makes it easy to get random numbers in any range. In the `getRandomInRange` function, I am creating a range from 1 to 5 then getting a random number from it. I add the value to the output through Kotlin's [String interpolation](https://kotlincompact.com/string-interpolation.html). You can add expressions as well as variables in strings.
```

Comment:
> In the `getRandomInRange` function, I am creating a range from 1 to 5 then getting a random number from it. I add the value to the output through Kotlin's

It is a better convention to use "We" instead of "I".

**In the `getRandomInRange` function, we are creating a range from 1 to 5 then getting a random number from it. We add the value to the output through Kotlin's**

Reply:
Done replacing I with we